packages/tools/client-debugger/client-debugger-view/src/components/ContainerSummaryView.tsx

```diff
@@ -26,140 +34,105 @@ initializeFluentUiIcons();
 /**
  * {@link ContainerSummaryView} input props.
  */
-export type ContainerSummaryViewProps = HasClientDebugger;
+export type ContainerSummaryViewProps = HasContainerId;
 
 /**
  * Debugger view displaying basic Container stats.
  */
 export function ContainerSummaryView(props: ContainerSummaryViewProps): React.ReactElement {
-	const { clientDebugger } = props;
-	const { container, containerNickname, containerId } = clientDebugger;
+	const { containerId } = props;
 
-	const myClientId = useMyClientId(clientDebugger);
-	const myClientConnection = useMyClientConnection(clientDebugger);
+	const messageRelay: IMessageRelay = useMessageRelay();
 
-	// #endregion
-
-	// #region Container State
-
-	const [containerAttachState, setContainerAttachState] = React.useState<AttachState>(
-		container.attachState,
-	);
-	const [containerConnectionState, setContainerConnectionState] = React.useState<ConnectionState>(
-		container.connectionState,
-	);
-	const [isContainerClosed, setIsContainerClosed] = React.useState<boolean>(container.closed);
+	const [containerState, setContainerState] = React.useState<
+		ContainerStateMetadata | undefined
+	>();
 
 	React.useEffect(() => {
-		function onContainerAttached(): void {
-			setContainerAttachState(container.attachState);
-		}
+		/**
+		 * Handlers for inbound messages related to the registry.
+		 */
+		const inboundMessageHandlers: InboundHandlers = {
+			["CONTAINER_STATE_CHANGE"]: (untypedMessage) => {
+				const message = untypedMessage as ContainerStateChangeMessage;
+				if (message.data.containerId === containerId) {
+					setContainerState(message.data.containerState);
+					return true;
+				}
+				return false;
+			},
+		};
 
-		function onContainerConnectionChange(): void {
-			setContainerConnectionState(container.connectionState);
+		/**
+		 * Event handler for messages coming from the webpage.
+		 */
+		function messageHandler(message: Partial<ISourcedDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: "ContainerSummaryView", // TODO: fix
+			});
 		}
 
-		function onContainerClosed(): void {
-			setIsContainerClosed(true);
-		}
+		messageRelay.on("message", messageHandler);
 
-		container.on("attached", onContainerAttached);
-		container.on("connected", onContainerConnectionChange);
-		container.on("disconnected", onContainerConnectionChange);
-		container.on("closed", onContainerClosed);
+		// Reset Container State data, to ensure we aren't displaying stats for the wrong container while
+		// we wait for a response from the new debugger.
+		// eslint-disable-next-line unicorn/no-useless-undefined
+		setContainerState(undefined);
```

Comment:
This is a good call. I think we should probably be doing that in our other `containerId`-dependent components as well for this same reason. Especially if the initial data fetch fails, we won't just continue displaying the wrong data to the user.

Reply:
True, I'll copy it to the rest of the views.